Add debugging steps for nest integration. (#37050)
home-assistant/home-assistant.github.io: source/_integrations/nest.markdown

```

- *Not receiving camera motion and person events*: assuming the integration is correctly configured (for example, the oauth and SDM API are set up correctly, you can see camera streams, and permissions are correctly set in [Partner Connections Manager](https://nestservices.google.com/partnerconnections)): If you are then still not seeing events, it's possible you need to adjust the Google Home App settings. Refer to the [Google Home App Notification Settings](#google-home-app-notification-settings) for details.

> - *Configuration error: Failed to create subscriber 'subscription/name' was not found*: By default Google pub/sub subscriptions will be deleted after 31 days of inactivity ([reference](https://cloud.google.com/knowledge/kb/pub-sub-subscriptions-disappeared-without-any-deletion-logs-000004170)). If this happens then the integration will fail, and you will see the preceding log line in your Home Assistant logs. If that is the case, then go to the [Device Access Console](https://console.nest.google.com/device-access/project-list) again, and re-enable the Pub/Sub Topic. The nest integration may need to be recreated to pick up the new pub/sub topic. You can optionally go to your [Google Cloud Console Pub/Sub subscription page](https://console.cloud.google.com/cloudpubsub/subscription/list), and edit the pub/sub subscription that was created by nest to not expire by default to avoid this issue in the future.

Reviewer: ```suggestion
- **Configuration error: Failed to create subscriber 'subscription/name' was not found** - By default, Google pub/sub subscriptions will be deleted after 31 days of inactivity ([reference](https://cloud.google.com/knowledge/kb/pub-sub-subscriptions-disappeared-without-any-deletion-logs-000004170)). If this happens, then the integration will fail, and you will see the preceding log line in your Home Assistant logs. If that is the case, then go to the [Device Access Console](https://console.nest.google.com/device-access/project-list) again, and re-enable the Pub/Sub Topic. The nest integration may need to be recreated to pick up the new pub/sub topic. To avoid this issue in the future, go to your [Google Cloud Console Pub/Sub subscription page](https://console.cloud.google.com/cloudpubsub/subscription/list) and edit the pub/sub subscription that was created by Nest. Change it so that is does not expire by default.
```

> - *Configuration error: Failed to create subscriber 'subscription/name' was not found*: By default Google pub/sub subscriptions will be deleted after 31 days of inactivity ([reference](https://cloud.google.com/knowledge/kb/pub-sub-subscriptions-disappeared-without-any-deletion-logs-000004170)). If this happens then the integration will fail, and you will see the preceding log line in your Home Assistant logs. If that is the case, then go to the [Device Access Console](https://console.nest.google.com/device-access/project-list) again, and re-enable the Pub/Sub Topic. The nest integration may need to be recreated to pick up the new pub/sub topic. You can optionally go to your [Google Cloud Console Pub/Sub subscription page](https://console.cloud.google.com/cloudpubsub/subscription/list), and edit the pub/sub subscription that was created by nest to not expire by default to avoid this issue in the future.

Reviewer: ```suggestion
- **Configuration error: Failed to create subscriber 'subscription/name' was not found** - By default, Google pub/sub subscriptions will be deleted after 31 days of inactivity ([reference](https://cloud.google.com/knowledge/kb/pub-sub-subscriptions-disappeared-without-any-deletion-logs-000004170)). If this happens, then the integration will fail, and you will see the preceding log line in your Home Assistant logs. If that is the case, go to the [Device Access Console](https://console.nest.google.com/device-access/project-list) again, and re-enable the Pub/Sub Topic. The Nest integration may need to be recreated to pick up the new pub/sub topic. to avoid this issue in the future, go to your [Google Cloud Console Pub/Sub subscription page](https://console.cloud.google.com/cloudpubsub/subscription/list), and edit the pub/sub subscription that was created by Nest. Change it so that it doesn't expire by default.
```

> - **Configuration error: Failed to create subscriber 'subscription/name' was not found** - By default, Google pub/sub subscriptions will be deleted after 31 days of inactivity ([reference](https://cloud.google.com/knowledge/kb/pub-sub-subscriptions-disappeared-without-any-deletion-logs-000004170)). If this happens, then the integration will fail, and you will see the preceding log line in your Home Assistant logs. If that is the case, then go to the [Device Access Console](https://console.nest.google.com/device-access/project-list) again, and re-enable the Pub/Sub Topic. The nest integration may need to be recreated to pick up the new pub/sub topic. To avoid this issue in the future, go to your [Google Cloud Console Pub/Sub subscription page](https://console.cloud.google.com/cloudpubsub/subscription/list) and edit the pub/sub subscription that was created by Nest. Change it so that is does not expire by default.

Reviewer: ```suggestion
- **Configuration error: Failed to create subscriber `subscription/name` was not found** - By default, Google pub/sub subscriptions will be deleted after 31 days of inactivity ([reference](https://cloud.google.com/knowledge/kb/pub-sub-subscriptions-disappeared-without-any-deletion-logs-000004170)). If this happens, then the integration will fail, and you will see the preceding log line in your Home Assistant logs. If that is the case, then go to the [Device Access Console](https://console.nest.google.com/device-access/project-list) again, and re-enable the Pub/Sub Topic. The nest integration may need to be recreated to pick up the new pub/sub topic. To avoid this issue in the future, go to your [Google Cloud Console Pub/Sub subscription page](https://console.cloud.google.com/cloudpubsub/subscription/list) and edit the pub/sub subscription that was created by Nest. Change it so that is does not expire by default.
```

> - **Configuration error: Failed to create subscriber `subscription/name` was not found** - By default, Google pub/sub subscriptions will be deleted after 31 days of inactivity ([reference](https://cloud.google.com/knowledge/kb/pub-sub-subscriptions-disappeared-without-any-deletion-logs-000004170)). If this happens, then the integration will fail, and you will see the preceding log line in your Home Assistant logs. If that is the case, then go to the [Device Access Console](https://console.nest.google.com/device-access/project-list) again, and re-enable the Pub/Sub Topic. The nest integration may need to be recreated to pick up the new pub/sub topic. To avoid this issue in the future, go to your [Google Cloud Console Pub/Sub subscription page](https://console.cloud.google.com/cloudpubsub/subscription/list) and edit the pub/sub subscription that was created by Nest. Change it so that is does not expire by default.

Reviewer: ```suggestion
- **Configuration error: Failed to create subscriber `subscription/name` was not found** - By default, Google pub/sub subscriptions will be deleted after 31 days of inactivity ([reference](https://cloud.google.com/knowledge/kb/pub-sub-subscriptions-disappeared-without-any-deletion-logs-000004170)). If this happens, then the integration will fail, and you will see the preceding log line in your Home Assistant logs. If that is the case, then:

  1. Go to the [Device Access Console](https://console.nest.google.com/device-access/project-list) and re-enable the Pub/Sub Topic
  2. The Nest integration may need to be recreated to pick up the new Pub/Sub topic
  3. (Optional) To prevent future expirations, visit your [Google Cloud Console Pub/Sub subscription page](https://console.cloud.google.com/cloudpubsub/subscription/list) and edit the Pub/Sub subscription created by Nest to not expire by default
```

```

- *Not receiving camera motion and person events*: assuming the integration is correctly configured (for example, the oauth and SDM API are set up correctly, you can see camera streams, and permissions are correctly set in [Partner Connections Manager](https://nestservices.google.com/partnerconnections)): If you are then still not seeing events, it's possible you need to adjust the Google Home App settings. Refer to the [Google Home App Notification Settings](#google-home-app-notification-settings) for details.

- **Configuration error: Failed to create subscriber `subscription/name` was not found** - By default, Google pub/sub subscriptions will be deleted after 31 days of inactivity ([reference](https://cloud.google.com/knowledge/kb/pub-sub-subscriptions-disappeared-without-any-deletion-logs-000004170)). If this happens, then the integration will fail, and you will see the preceding log line in your Home Assistant logs. If that is the case, then:

  1. Go to the [Device Access Console](https://console.nest.google.com/device-access/project-list) and re-enable the Pub/Sub Topic.

>   2. The Nest integration may need to be recreated to pick up the new Pub/Sub topic.

Reviewer 2: Thanks for adding this.  I think we may need to add a repair flow or reconfigure flow for this.

>   1. Go to the [Device Access Console](https://console.nest.google.com/device-access/project-list) and re-enable the Pub/Sub Topic
>   2. The Nest integration may need to be recreated to pick up the new Pub/Sub topic
>   3. (Optional) To prevent future expirations, visit your [Google Cloud Console Pub/Sub subscription page](https://console.cloud.google.com/cloudpubsub/subscription/list) and edit the Pub/Sub subscription created by Nest to not expire by default

Reviewer: ```suggestion
  1. Go to the [Device Access Console](https://console.nest.google.com/device-access/project-list) and re-enable the Pub/Sub Topic.
  2. The Nest integration may need to be recreated to pick up the new Pub/Sub topic.
  3. (Optional) To prevent future expirations, visit your [Google Cloud Console Pub/Sub subscription page](https://console.cloud.google.com/cloudpubsub/subscription/list) and edit the Pub/Sub subscription created by Nest to not expire by default.
```

>   3. (Optional) To prevent future expirations, visit your [Google Cloud Console Pub/Sub subscription page](https://console.cloud.google.com/cloudpubsub/subscription/list) and edit the Pub/Sub subscription created by Nest to not expire by default.

Reviewer: ```suggestion
  3. (Optional) To prevent future expirations, go to your [Google Cloud Console Pub/Sub subscription page](https://console.cloud.google.com/cloudpubsub/subscription/list) and edit the Pub/Sub subscription created by Nest to not expire by default.
```
